openllm-python/src/openllm/_llm.py

```diff
@@ -988,14 +988,16 @@ def generate_iterator(self,
       past_key_values = out = token = None
       finish_reason = None
       for i in range(config['max_new_tokens']):
-        torch.cuda.synchronize()
+        if torch.cuda.is_available():
+          torch.cuda.synchronize()
```

Comment:
You can make this one line

and lets make a separate PR for this. This doesn't seem related at all.

Reply:
OK no problem.